test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -1310,6 +1319,32 @@ func getPoolNodes(f *framework.Framework, poolName string) []*v1.Node {
 	return nodes
 }
 
+// getPoolInitialSize returns the initial size of the node pool taking into
+// account that it may span multiple zones. In that case, node pool consists of
+// multiple migs all containing initialNodeCount nodes.
+func getPoolInitialSize(poolName string) int {
```

Comment:
I think I understand what this function does, but is there any reason we can't use `getPoolSize` instead? Since we create node-pool synchronously using gcloud command, nodes should register even before the operation ends, so they'd already be there.

Reply:
What if we hit quota in the test and only one out of three nodes registers? We will probably get error from gcloud.

Hm, I guess this would indeed work, but does that mean all our WaitForReadyNodes after adding node pools are completely redundant? (see for example line 689).

Reply:
Good point. Let me check this - I think it was changed some time ago from using API calls (asynchronous), so this may be now redundant. 

A couple of minutes (I've observed ~1.5m) after a new node-pool is added, GKE may resize master, but I don't think those waits protect against this as they would usually succeed before the resize is triggered (as would get nodes request)

Reply:
I don't like the idea of removing those waits in this PR, since I am changing a lot of other things. So how about I put this PR on hold, remove the waits from the tests in another PR, watch if the tests are healthy and then submit this one with the changes you suggested?

Reply:
Sure, didn't mean to ask you to do more than intended in this PR! We can also maybe merge this, and create an issue for investigating/cleaning up those waits?

Reply:
Follow-up tracked in kubernetes/autoscaler#666